### ISSUE TITLE
Collab Migration Fail Bug

### ISSUE THREAD

**Issue by robertsonbrinker** [Clear-Bible/ClearDashboard#605]
TL;DR There's probably something wrong with the parallelcorpus-verse migration but I don't think I'd be able to spot what it is.

When loading up a 0.4.6.1 or earlier project using the `collab` branch the migrations fail.
![image](https://user-images.githubusercontent.com/60048070/228965506-700f17ca-7664-4fab-b602-3e7b088246d3.png)
Specifically when applying the first migration that needs to be applied to a project of that version number (which is the ` parallelcorpus-verse.cs` migration) it fails on the 4th SQL command.  (on older versions of this branch it failed on the 3rd SQL command).

1. 

> ALTER TABLE "Verse" ADD "ParallelCorpusId" TEXT NOT NULL DEFAULT '00000000-0000-0000-0000-000000000000';

2. 

> CREATE INDEX "IX_Verse_ParallelCorpusId" ON "Verse" ("ParallelCorpusId");

3. 

> CREATE TABLE "ef_temp_Verse" (
>     "Id" TEXT NOT NULL CONSTRAINT "PK_Verse" PRIMARY KEY,
>     "BookNumber" INTEGER NULL,
>     "ChapterNumber" INTEGER NULL,
>     "CorpusHistoryId" TEXT NULL,
>     "CorpusId" TEXT NULL,
>     "Created" INTEGER NOT NULL,
>     "ParallelCorpusId" TEXT NOT NULL,
>     "UserId" TEXT NOT NULL,
>     "VerseBBBCCCVVV" TEXT NULL,
>     "VerseMappingId" TEXT NOT NULL,
>     "VerseNumber" INTEGER NULL,
>     "VerseText" TEXT NULL,
>     CONSTRAINT "FK_Verse_CorpusHistory_CorpusHistoryId" FOREIGN KEY ("CorpusHistoryId") REFERENCES "CorpusHistory" ("Id"),
>     CONSTRAINT "FK_Verse_Corpus_CorpusId" FOREIGN KEY ("CorpusId") REFERENCES "Corpus" ("Id"),
>     CONSTRAINT "FK_Verse_ParallelCorpus_ParallelCorpusId" FOREIGN KEY ("ParallelCorpusId") REFERENCES "ParallelCorpus" ("Id") ON DELETE CASCADE,
>     CONSTRAINT "FK_Verse_User_UserId" FOREIGN KEY ("UserId") REFERENCES "User" ("Id") ON DELETE CASCADE,
>     CONSTRAINT "FK_Verse_VerseMapping_VerseMappingId" FOREIGN KEY ("VerseMappingId") REFERENCES "VerseMapping" ("Id") ON DELETE CASCADE
> );

4. (Fails on this one)

>  INSERT INTO "ef_temp_Verse" ("Id", "BookNumber", "ChapterNumber", "CorpusHistoryId", "CorpusId", "Created", "ParallelCorpusId", "UserId", "VerseBBBCCCVVV", "VerseMappingId", "VerseNumber", "VerseText")
> SELECT "Id", "BookNumber", "ChapterNumber", "CorpusHistoryId", "CorpusId", "Created", "ParallelCorpusId", "UserId", "VerseBBBCCCVVV", "VerseMappingId", "VerseNumber", "VerseText"
> FROM "Verse";

When running that SQL command the Foreign Key Error originates in the location marked in the picture below.  
![image](https://user-images.githubusercontent.com/60048070/228965953-a37771ce-af25-4804-a1ef-b9443b519bcb.png)

So then after the migration fails then the `LoadProjectSlice` fails to obtain the `currentproject`.  
![image](https://user-images.githubusercontent.com/60048070/228966162-e11914ef-f5e3-45fa-aafb-25524247ea4d.png)
This is presumably because the migration failed to add the `LastMergedCommitSha` column to the project, but I'm not sure why this column, as opposed to any other, causes this issue.  Is it because this is the only project column that was added when working on `collab` functionality?  

When I remove any mentions of the `LastMergedCommitSha` from the solution this error does not pop up.  

In conclusion: Is there something wrong with the parallelcorpus-verse migration?  Sounds like it, but I don't think I'd be able to spot what it is.
